pinot-controller/src/main/java/org/apache/pinot/controller/recommender/rules/RulesToExecute.java

```diff
@@ -45,6 +46,8 @@
   public static class RuleFactory {
     public static AbstractRule getRule(Rule rule, InputManager inputManager, ConfigManager outputManager) {
       switch (rule) {
+        case SegmentSizeRule:
```

Comment:
I believe adding here will also make sure it will be executed first ?

Reply:
Not in the switch statement. The order of rules is defined in `Rule` enum. In `RecommenderDriver.run()`, it iterates over `Rule.values()` which returns the enum values in the defined order.